redisgraph/graph.py

```diff
@@ -121,16 +144,37 @@ def query(self, q, params=None, timeout=None):
         """
         Executes a query against the graph.
         """
+
+        # maintain original 'q'
+        query = q
+
+        # handle query parameters
         if params is not None:
-            q = self.build_params_header(params) + q
+            query = self.build_params_header(params) + query
 
-        command = ["GRAPH.QUERY", self.name, q, "--compact"]
+        # construct query command
+        # ask for compact result-set format
+        # specify known version
+        command = ["GRAPH.QUERY", self.name, query, "--compact", "version", self.version]
```

Comment:
use `version` only when server supports this feature, for older version of RedisGraph this will trigger arity error